library/core/src/iter/traits/iterator.rs

```diff
@@ -48,13 +48,24 @@ pub trait Iterator {
     type Item;
 
     /// Advances the iterator and returns the next value.
+    /// Returns [`None`] when iteration is finished.
     ///
-    /// Returns [`None`] when iteration is finished. Individual iterator
-    /// implementations may choose to resume iteration, and so calling `next()`
-    /// again may or may not eventually start returning [`Some(Item)`] again at some
-    /// point.
+    /// # Behavior after [`None`] is returned
+    ///
+    /// Calling `next()` on an iterator after it has returned `None`
+    /// is in general a perfectly valid use of the interface.
+    /// Implementations may choose to resume iteration, and so calling `next()`
+    /// again may eventually start returning [`Some(Item)`] again at some
+    /// point. Alternately, implementations may choose to continue returning [`None`]
+    /// forever, in which case they should implement [`FusedIterator`] as well.
```

Comment:
I wonder if we should spin a rustdoc feature request out of this too. Rustdoc already notes when a type is not Send/Sync, and when a trait is not object-safe. We could put it in charge of pointing out in some appropriately obtrusive way when an Iterator is not FusedIterator, with a sentence or two on what this means, or a link.

<img src="https://github.com/user-attachments/assets/87b0cfad-a4c2-4fe4-9c26-6200d00a4923" width="300">